package.json

```diff
@@ -6,19 +6,28 @@
     "@gi-types/gjs-environment": "^1.1.0",
     "@gi-types/gtk4": "^4.6.1",
     "@gi-types/gtk4-types": "^1.0.0",
-    "@rollup/plugin-commonjs": "^22.0.1",
-    "@rollup/plugin-node-resolve": "^13.3.0",
+    "@rollup/plugin-commonjs": "^25.0.4",
+    "@rollup/plugin-node-resolve": "^15.2.1",
     "events": "^3.3.0",
+    "husky": "^8.0.3",
+    "lint-staged": "^14.0.1",
     "ltx": "git://github.com/xmppjs/ltx.git#093c7bdced57c22f7eccba3bccfc141150654395",
-    "postcss": "^8.4.14",
-    "prettier": "^2.7.1",
-    "rollup": "^2.76.0",
+    "postcss": "^8.4.28",
+    "prettier": "^3.0.2",
+    "rollup": "^3.28.1",
     "rollup-plugin-ignore": "^1.0.10",
     "rollup-plugin-node-polyfills": "^0.2.1",
-    "rome": "^12.1.1"
+    "rome": "^12.1.3"
   },
   "type": "module",
   "scripts": {
-    "postinstall": "rollup -c rollup.config.js"
+    "postinstall": "rollup -c rollup.config.js",
+    "prepare": "husky install"
+  },
+  "lint-staged": {
+    "*.{js,cjs,mjs}": "rome format --write",
+    "*.{css,json,md}": "prettier --write",
+    "*.{rs}": "flatpak run --command=/usr/lib/sdk/rust-stable/bin/rustfmt --file-forwarding org.gnome.Sdk//45beta @@ ",
+    "*.{vala,c,h}": "flatpak run --command=/usr/lib/sdk/vala/bin/uncrustify --file-forwarding org.gnome.Sdk//45beta -c - --replace --no-backup --set indent_with_tabs=0 --set nl_end_of_file=force --set nl_end_of_file_min=1 --set indent_columns=2 @@ "
```

Comment:
Looks about right. I don't have experience with the uncrustify's vala code prettifier though. But if it will behave incorrectly, we can still enable it later. So far it looks good.

Reply:
Yeah, it's hard to say until it's being run on real code. Personally I prefer `clang-format` to uncrustify, but GNOME's code style for C is hard to express for any formatter.

Agreed that we can always change/tweak it later.

Reply:
For clang-format and C that could be used: https://gitlab.gnome.org/GNOME/gtk/-/blob/main/.clang-format

Reply:
Sure, that works. I mean, I'm not exaggerating when I say it takes *a lot* of work to get it to match GNOME's C style, but I'm not worried too much about it :shrug:

Reply:
There are only 2 C files. reformatting them shouldn't be a problem. I guess nobody here is very attached to using the GNOME style.